Add examples and update constraints for Kafka API (#13492)
ydb-platform/ydb: ydb/docs/en/core/reference/kafka-api/auth.md

# Kafka API Authentication

## Enabling authentication

> Authentication is always enabled when using the [Kafka API in Yandex Cloud](https://yandex.cloud/docs/data-streams/kafkaapi/auth).

Reviewer: Documentation about Yandex Cloud is published on https://yandex.cloud/ru/docs instead of here.

> When you run YDB cluster by your own, [anonymous authentication](../../security/authentication.md#anonymous) is used by default.

Reviewer 2: ```suggestion
If you deployed [a single-node local YDB cluster](../../quickstart.md), [anonymous authentication](../../security/authentication.md#anonymous) is used by default.
```

It doesn't require a username and password.

> To require authentication see [Authentication](../../security/authentication#static-credentials).

Reviewer: ```suggestion
To require authentication see [Authentication](../../security/authentication.md#static-credentials).
```

> Authentication is always enabled when using the [Kafka API in Yandex Cloud](https://yandex.cloud/en/docs/data-streams/kafkaapi/auth).
> 
> When you run [a single-node local {{ ydb-short-name }} cluster](../../quickstart.md), [anonymous authentication](../../security/authentication.md#anonymous) is used by default.
> It doesn't require a username and password.
> To require authentication see [Authentication](../../security/authentication.md#static-credentials).

Reviewer: ```suggestion
When you run [a single-node local {{ ydb-short-name }} cluster](../../quickstart.md), [anonymous authentication](../../security/authentication.md#anonymous) is used by default. It doesn't require a username and password.

To require authentication see [Authentication](../../security/authentication.md#static-credentials).

Authentication is always enabled when using the [Kafka API in Yandex Cloud](https://yandex.cloud/en/docs/data-streams/kafkaapi/auth).
```

> To require authentication, set the [`enforce_user_token_requirement` configuration option](../configuration/index.md#auth).

Reviewer: If the Kafka API is controlled by the same settings as the YDB-native API, I'd say exactly that and then give a link to https://ydb.tech/docs/en/security/authentication instead of trying to retell it here.

> ## How does authentication work in the Kafka API?
> 
> The Kafka API uses the SASL_PLAINTEXT/PLAIN or SASL_SSL/PLAIN authentication mechanism.
> 
> The following variables are required for authentication:
> 
> * `<user-name>` — the username. For information about user management, refer to the section [Authorization](../../security/authorization.md#user).
> * `<password>` — the user's password. For information about user management, refer to the section [Authorization](../../security/authorization.md#user).
> * `<database>` — [the database path](../../concepts/connect#database).

Reviewer: ```suggestion
## How does authentication work in the Kafka API?

The Kafka API uses the `SASL_PLAINTEXT/PLAIN` or `SASL_SSL/PLAIN` authentication mechanism.

The following variables are required for authentication:

* `<user-name>` — the username. For information about user management, refer to the [Authorization](../../security/authorization.md#user) section.
* `<password>` — the user's password. For information about user management, refer to the [Authorization](../../security/authorization.md#user) section.
* `<database>` — [the database path](../../concepts/connect.md#database).
```

> These parameters form the following variables you can use in `sasl.jaas.config` kafka client property:
> 
> * `<sasl.username> = <user-name>@<database>`
> * `<sasl.password> = <password>`
> 
> {% note info %}
> 
> The `<sasl.username>` and `<sasl.password>` parameters are formed differently. See [examples](./examples#authentication-in-cloud-examples) for details.
> 
> {% endnote %}
> 
> For authentication examples, see [Kafka API Usage examples](./examples.md).

Reviewer: ```suggestion
These parameters form the following variables, which you can use in the `sasl.jaas.config` Kafka client property:

* `<sasl.username> = <user-name>@<database>`
* `<sasl.password> = <password>`

{% note info %}

The `<sasl.username>` and `<sasl.password>` parameters are formed differently. See [examples](./examples#authentication-in-cloud-examples) for details.

{% endnote %}

For authentication examples, see [Kafka API usage examples](./examples.md).
```

# Kafka API Authentication

## Enabling authentication

Authentication is always enabled when using the [Kafka API in Yandex Cloud](https://yandex.cloud/docs/data-streams/kafkaapi/auth).

When you run YDB cluster [by your own](../../quickstart.md), [anonymous authentication](../../security/authentication.md#anonymous) is used by default.
It doesn't require a username and password.
To require authentication, set the [`enforce_user_token_requirement` configuration option](../configuration/index.md#auth).

## How does authentication work in the Kafka API?

> In the Kafka API authentication is conducted through the SASL_PLAINTEXT/PLAIN or SASL_SSL/PLAIN mechanisms.

Author: Добавлю
